pilot/pkg/status/state.go

```diff
@@ -162,41 +171,28 @@ func (c *DistributionController) initK8sResource(gvr schema.GroupVersionResource
 		return result
 	}
 	result = c.dynamicClient.Resource(gvr)
+	c.mu.Lock()
 	c.knownResources[gvr] = result
+	c.mu.Unlock()
 	return
 }
 
-func (c *DistributionController) writeStatus(ctx context.Context, config Resource, distributionState Progress) {
-	// Note: I'd like to use Pilot's ConfigStore here to avoid duplicate reads and writes, but
-	// the update() function is not implemented, and the Get() function returns the resource
-	// in a different format than is needed for k8s.updateStatus.
-	c.currentlyWriting.Lock(config)
-	defer c.currentlyWriting.Unlock(config)
-	resourceInterface := c.initK8sResource(config.GroupVersionResource).
-		Namespace(config.Namespace)
-	// should this be moved to some sort of InformerCache for speed?
-	current, err := resourceInterface.Get(ctx, config.Name, metav1.GetOptions{ResourceVersion: config.ResourceVersion})
-	if err != nil {
-		if errors.IsGone(err) || errors.IsNotFound(err) {
-			// this resource has been deleted.  prune its state and move on.
-			c.pruneOldVersion(config)
-			return
-		}
-		scope.Errorf("Encountered unexpected error when retrieving status for %v: %s", config, err)
-		return
-
-	}
-	if config.ResourceVersion != current.GetResourceVersion() {
+func (c *DistributionController) writeStatus(config Resource, distributionState Progress) {
+	schema, _ := collections.All.FindByGroupVersionResource(config.GroupVersionResource)
+	current := c.configStore.Get(schema.Resource().GroupVersionKind(), config.Name, config.Namespace)
```

Comment:
this can give a nil pointer exception, can we handle `FindByGroupVersionResource` failing?

Reply:
I guess in practice it can never happen, but would rather log an error than panic if it does